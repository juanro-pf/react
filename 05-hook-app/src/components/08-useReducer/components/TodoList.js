import React from 'react';
import { TodoListItem } from './TodoListItem';
import PropTypes from 'prop-types';

export const TodoList = ({todos, handleToggle, handleDelete}) => {
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map( (todo, i) => 
                    <TodoListItem
                        key= {todo.id}
                        todo= {todo}
                        i= {i}
                        handleToggle= {handleToggle}
                        handleDelete= {handleDelete}
                    />
                )
            }
        </ul>
    );
};

TodoList.propTypes= {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
};