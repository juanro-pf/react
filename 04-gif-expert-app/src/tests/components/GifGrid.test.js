import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas a GifGrid', () => {
        
    const category= 'Batman';

    test('Snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper= shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('mostar items al cargar imagenes con useFetchGifs', () => {
        
        const imgs= [{
            id: 'ABC',
            url: 'http://asd.jpg',
            title: 'Title'
        },
        {
            id: 'ABCD',
            url: 'http://asd.jpg',
            title: 'Title'
        }];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });

        const wrapper= shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);

    });    
    
});
