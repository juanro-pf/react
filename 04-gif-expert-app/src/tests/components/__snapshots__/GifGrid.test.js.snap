// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifGrid Snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
